main/SQLHelper.cpp

```diff
@@ -6017,9 +6066,32 @@ void CSQLHelper::UpdateTemperatureLog()
 			{
 			case pTypeRego6XXTemp:
 			case pTypeTEMP:
-			case pTypeThermostat:
 				temp = static_cast<float>(atof(splitresults[0].c_str()));
 				break;
+			case pTypeThermostat:
+				{
+
+					const std::string& HwdId = sd[7];
+					//get virtual thermostat hardware
+					CDomoticzHardwareBase* Hardware = m_mainworker.GetHardwareByIDType( HwdId, HTYPE_VirtualThermostat);
+					//if not a virtual thermostat
+					if (Hardware==nullptr)
+					{
+						temp = static_cast<float>(std::stof(splitresults[0].c_str()));
+					}
+					else
+					{
+						//for virtual thermostat record Room Temperature / set point & power level 0..100%
+						//set point temperature record as chill
+						std::string Options  = sd[6];
+						chill = static_cast<float>(std::stof(splitresults[0].c_str()));
+						//record power % as humidity
+						humidity = std::stoi(VirtualThermostatGetOption("Power", Options).c_str());
+						//record room temp 
+						temp = (float)std::stof(VirtualThermostatGetOption("RoomTemp", Options).c_str());
```

Comment:
you do not need .c_str() here, also the cast to float is not needed (std::stof is returning a float)

Reply:
Ok done.